### ISSUE TITLE
Non-standard fields ignored by the BibTeX parser

### ISSUE THREAD

**Issue by hubgit** [dsifford/astrocite#12]
When parsing a BibTeX item, only the standard BibTeX fields (those listed in the `FIELD_MAP` Map) are preserved. This loses useful information such as the DOI, ISSN and URL.

Would it be possible to either add an option to include non-standard fields in the mapping or allow the `FIELD_MAP` to be extended as a parse option?

`url` and `doi` are both mentioned in the `verbatimProperties` array so are already parsed correctly, but are then dropped during the field mapping.

**Code**

```js
const {
    parse
} = require('astrocite-bibtex');

const data = `@article{Verkhratsky2019Evolution,
	journal = {Advances in experimental medicine and biology},
        doi = {10.1007/978-981-13-9913-8_2},
        url = {https://link.springer.com/chapter/10.1007%2F978-981-13-9913-8_2},
	issn = {0065-2598},
	pmid = {31583583},
	address = {United States},
	title = {Evolution of Neuroglia},
	volume = {1175},
	author = {Verkhratsky, Alexei and Ho, Margaret S and Parpura, Vladimir},
	pages = {15--44},
	date = {2019},
	year = {2019},
}`;

const item = parse(data);

console.log(item);
```

**Expected behavior:**

```
[
  {
    id: 'Verkhratsky2019Evolution',
    type: 'article',
    'container-title': 'Advances in experimental medicine and biology',
    'publisher-place': 'United States',
    title: 'Evolution of Neuroglia',
    volume: '1175',
    author: [ [Object], [Object], [Object] ],
    page: '15–44',
    issued: { 'date-parts': [Array] },
    URL: 'https://link.springer.com/chapter/10.1007%2F978-981-13-9913-8_2',
    DOI: '10.1007/978-981-13-9913-8_2',
    ISSN: '0065-2598',
    PMID: '31583583'
  }
]
```

**Actual behavior:**

```
[
  {
    id: 'Verkhratsky2019Evolution',
    type: 'article',
    'container-title': 'Advances in experimental medicine and biology',
    'publisher-place': 'United States',
    title: 'Evolution of Neuroglia',
    volume: '1175',
    author: [ [Object], [Object], [Object] ],
    page: '15–44',
    issued: { 'date-parts': [Array] }
  }
]```

**Comment by hubgit** [comment]
https://www.acm.org/publications/authors/bibtex-formatting has an example BibTeX file containing these extra fields:

```bibtex
@article{Abril:2007:PHD:1188913.1188915,
author = {Abril, Patricia S. and Plant, Robert},
title = {The Patent Holder's Dilemma: Buy, Sell, or Troll?},
journal = {Commun. ACM},
issue_date = {January 2007},
volume = {50},
number = {1},
month = jan,
year = {2007},
issn = {0001-0782},
pages = {36--44},
numpages = {9},
url = {https://doi.org/10.1145/1188913.1188915},
doi = {10.1145/1188913.1188915},
acmid = {1188915},
publisher = {ACM},
address = {New York, NY, USA},
}
```

**Comment by dsifford** [comment]
Ping @retorquere

**Comment by retorquere** [comment]
I have a fix in #15, but this is not really the code I'm most familiar with -- I've done work on the bibtex parser, but it was the parsed-bibtex-to-CSL part that needed a change. The latter part I'm only superficially familiar with, as I don't use it myself, I just use the parser (for parsed-bibtex-to-Zotero conversions).

**Comment by retorquere** [comment]
(also, bibtex is so incredibly flexible that there's samples for all sorts of really, really odd behaviors -- my own converter addresses more than this in [the stuff I found over the years](https://github.com/retorquere/zotero-better-bibtex/blob/master/translators/Better%20BibTeX.ts#L302) but that's guaranteed not comprehensive)
